### ISSUE TITLE
Nesting two levels deep makes route inaccessible

### ISSUE THREAD

**Issue by jvtrigueros** [kumarshantanu/calfpath#13]
I started with a very basic example from the README and added one more level of nesting and was not able to get it work, unsure if a bug or misconfiguration. Here's the code:

```clojure
(ns calfpath-routes.core
  (:require [calfpath.route :as r]
            [ring.mock.request :as mock])
  (:gen-class))

(defn app-routes
  "Return a vector of route specs."
  []
  [{:uri    "/v1*"
    :nested [{:uri    "/cart*"
              :nested [{:uri     "/item/:id"
                        :method  :get
                        :handler (fn [request]
                                   {:status 200
                                    :body   (:id request)})}]}]}])

;; create a Ring handler from given routes
(def ring-handler
  (-> (app-routes)
      (r/compile-routes)
      r/make-dispatcher))

(comment
  (println
    (:body (ring-handler
             (mock/request :get "/v1/cart/item/10")))))
```

The print statements returns:
```
400 Bad request. URI does not match any available uri-template.

Available URI templates:
/v1/cart/item/:id
```
If I re-write the routes like so:
```
{:uri    "/v1*"
    :nested [{:uri     "/cart/item/:id"
              :method  :get
              :handler (fn [request]
                         {:status 200
                          :body   (:id request)})}]}
```
Then the route works as expected.

**Comment by kumarshantanu** [comment]
@jvtrigueros Thanks for reporting this - I can reproduce it in version `0.7.2` but not in `0.8.0-alpha1`. Could you please try the alpha and let me know if it works for you?

**Comment by kumarshantanu** [comment]
Fixed in version `0.8.0`.
